contributors/design-proposals/controller_history.md

```diff
@@ -0,0 +1,447 @@
+# Controller History
+
+**Author**: kow3ns@
+
+**Status**: Proposal
+
+## Abstract 
+In Kubernetes, in order to update and rollback the configuration and binary 
+images of controller managed Pods, users mutate DaemonSet, StatefulSet, 
+and ReplicaSet (via Deployment) Objects, and the corresponding controllers 
```

Comment:
We don't mutate ReplicaSets*, we create new copies when the Deployment is mutated.

* apart from syncing annotations and spec.MinReadySeconds between the two objects